hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -2049,9 +2060,20 @@ def _pooled_kwargs(self, ir_type, kwargs):
 
     def _pop_ir_tree_node(self, ir_type: IRTypeName, kwargs: IRKWargsType) -> IRNode:
         assert self.ir_tree_nodes is not None
+
+        if self.ir_tree_nodes == []:
+            self.mark_overrun()
+
         node = self.ir_tree_nodes.pop(0)
-        assert node.ir_type == ir_type
-        assert kwargs == node.kwargs
+        if node.ir_type != ir_type or node.kwargs != kwargs:
+            # Unlike buffers, not every ir tree is a valid choice sequence. If
+            # we want to maintain determinism we don't have many options here
+            # beyond giving up when a modified tree becomes misaligned.
+            #
+            # For what it's worth, misaligned buffers — albeit valid — are rather
+            # unlikely to be *useful* buffers, so this isn't an enormous
+            # downgrade.
+            self.mark_overrun()
```

Comment:
I think what we care about is that `node.value` is allowed by the arguments to this function - if the kwargs don't match exactly that seems OK to me?

We could even be somewhat flexible about converting between integer-values int/float (/bool?) and str/bytes (via utf-8).

Reply:
hmm, I guess we could do this. I don't know how useful it would be. I'd love to empirically evaluate misalignments on real properties one day 😄